chain-crypto/src/algorithms/vrf/vrf.rs

```diff
@@ -160,27 +176,30 @@ impl PublicKey {
 }
 
 impl ProvenOutputSeed {
-    pub const BYTES_LEN: usize = dleq::Zkp::BYTES_LEN + GroupElement::BYTES_LEN;
+    pub const BYTES_LEN: usize = dleq::Proof::PROOF_SIZE + GroupElement::BYTES_LEN;
     /// Verify a proof for a given public key and a data slice
     pub fn verify(&self, public_key: &PublicKey, input: &[u8]) -> bool {
-        self.dleq_proof.verify(
-            &GroupElement::generator(),
-            &GroupElement::from_hash(&input),
-            &public_key.0,
-            &self.u.0,
-        )
+        let dleq = dleq::Dleq {
+            g1: &GroupElement::generator(),
+            h1: &public_key.0,
+            g2: &GroupElement::from_hash(&input),
+            h2: &self.u.0,
+        };
+        dleq::verify(&dleq, &self.dleq_proof)
     }
 
     pub fn to_buffer(&self, output: &mut [u8]) {
         assert_eq!(output.len(), Self::BYTES_LEN);
         output[0..GroupElement::BYTES_LEN].copy_from_slice(&self.u.0.to_bytes());
         self.dleq_proof
-            .write_to_bytes(&mut output[GroupElement::BYTES_LEN..]);
+            .to_bytes(&mut output[GroupElement::BYTES_LEN..]);
```

Comment:
Also, we got the `to_*` functions renamed for keeping consistency across crates.